src/pages/Search/components/Heatmap.js

```diff
@@ -12,36 +16,56 @@ import {
   BoxRow,
   TimeZone,
 } from './Heatmap.styles';
-import Box from './Box';
 
 const Heatmap = ({ posts }) => {
-  const rows = posts.map((el) => {
-    const columns = el.map((totalPosts) => (
-      <Box numTotalPosts={totalPosts.length} />
-    ));
-    return <BoxRow key={el}>{columns}</BoxRow>;
+  const [selected, setSelected] = useState('');
+  const [timePosts, setTimePosts] = useState([]);
+
+  const handleActive = (key) => setSelected(key);
+
+  const setPosts = (selectedPosts) => setTimePosts(selectedPosts);
+
+  const rows = posts.map((el, elIndex) => {
+    const columns = el.map((totalPosts, i) => {
+      const { length } = totalPosts;
+      const key = `${elIndex}-${length}-${i}`;
+      return (
+        <Box
+          key={key}
+          index={key}
+          selected={selected === key}
+          handleActive={handleActive}
```

Comment:
you can use `setSelected` here directly and remove the `handleActive` above. Same goes for `setPosts`, use `setTimePosts` directely